src/repositories/QueueRepository/__tests__/QueueRepository.int.spec.ts

```diff
@@ -208,4 +209,40 @@ describe("Queue Repository tests", () => {
     expect(DateTime.fromJSDate(playerInDb?.queueTime!).toISO()).toEqual(mockBallChaser.queueTime.toISO());
     expect(playerInDb?.team).toBeNull();
   });
+  describe("check if player is in queue", () => {
+    it("player is in queue", async () => {
+      const mockBallChaser = BallChaserQueueBuilder.single();
+      await manuallyAddBallChaserToQueue(mockBallChaser);
+
+      const playerInQueue = await QueueRepository.isPlayerInQueue(mockBallChaser.id);
+
+      expect(playerInQueue).toEqual(true);
+    });
+    it("player is not in queue", async () => {
+      const mockBallChaser = BallChaserQueueBuilder.single();
+      await manuallyAddBallChaserToQueue(mockBallChaser);
+
+      const playerNotInQueue = await QueueRepository.isPlayerInQueue(faker.datatype.uuid());
+
+      expect(playerNotInQueue).toEqual(false);
+    });
+  });
+  describe("check if player is team captain", () => {
+    it("player is team captain", async () => {
+      const mockBallChaser = BallChaserQueueBuilder.single({ team: Team.Blue });
+      await manuallyAddBallChaserToQueue(mockBallChaser);
+
+      const isCaptain = await QueueRepository.isTeamCaptain(mockBallChaser.id, Team.Blue);
+
+      expect(isCaptain).toEqual(true);
+    });
+    it("player is not team captain", async () => {
+      const mockBallChaser = BallChaserQueueBuilder.single({ team: Team.Blue });
+      await manuallyAddBallChaserToQueue(mockBallChaser);
+
+      const isNotCaptain = await QueueRepository.isTeamCaptain(faker.datatype.uuid(), Team.Orange);
```

Comment:
Right now you have a test to make sure the captain is found, and that the captain is not found when both the id and team color are incorrect. What about separately? What if the id is correct, but the team is wrong? Or the team is correct, but the id is wrong? Do those work? If you have those two, then you don't need the wrong id and wrong team test.

Does that logic make sense? There are 3 branches here, so we need a test for each branch.

Reply:
Added a couple of tests